File: js/complete/no020.js
#!/usr/bin/env node

/* Find the sum of the digits in the number 100! */

var fns = require('../functions.js'),
    operator = require('../operator.js'),
    timer = require('../timer.js');

exports.main = function(verbose) {
    if (typeof verbose == 'undefined') {
        verbose = false;
    }
    // Sum will cast to Number
    return operator.sum(fns.factorial(100, true).split(''));
};

if (require.main === module) {
    timer.timer(20, exports.main);
}

File: python_code/complete/no020.py
#!/usr/bin/env python

# Find the sum of the digits in the number 100!

from math import factorial

from python_code.decorators import euler_timer

def main(verbose=False):
    return sum([int(digit) for digit in str(factorial(100))])

if __name__ == '__main__':
    print euler_timer(20)(main)(verbose=True)
